### PR TITLE
updating test plan for Labels

### PR DESCRIPTION
**Pull request by eriktate**
Adding a line for resource-based labels in the `Labels` section

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/testplan.md [hunk 1]
```diff
@@ -20,6 +20,7 @@ as well as an upgrade of the previous version of Teleport.
 - [ ] Labels
   - [ ] Static Labels
   - [ ] Dynamic Labels
+  - [ ] Resource-based Labels
```
Comment (marcoandredinis) [inline .github/ISSUE_TEMPLATE/testplan.md:23]: What are resource based labels? Is it the labels we get from the compute provider the instance is running in? Like AWS EC2 tags and Azure VM labels?

**Review by marcoandredinis — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/testplan.md (hunk 1 above)
Comment (eriktate) [reply .github/ISSUE_TEMPLATE/testplan.md:23]: They're labels that can be created at run time with a `server_info` resource: https://goteleport.com/docs/admin-guides/management/admin/labels/#apply-resource-based-labels

Comment (marcoandredinis) [reply .github/ISSUE_TEMPLATE/testplan.md:23]: Thank you 🙏 
I've used `server_info` but was not aware of the "resource based labels"

Comment (bl-nero) [reply .github/ISSUE_TEMPLATE/testplan.md:23]: We can retain the name, but can we please make this element a link to that document? Seeing "server_info labels", I also wouldn't know what it is.

Comment (eriktate) [reply .github/ISSUE_TEMPLATE/testplan.md]: I added the docs link and also mentioned that it uses `server_info` inline :+1:

**Review by rosstimothy — APPROVED** [review]
Thanks @eriktate!

**Pull request merged**